openfifth/aspen-discovery
# Usage graphs and CSV: Aspen API, SideLoads, and Axis360 (Boundless)

This patch adds three sets of usage graphs, raw usage data tables, and CSV download. While there are more usage graphs and CSV downloads to be added, I thought it best to break down what would have been one massive PR into smaller ones. 

The following sections are affected: 
- ILS (refactor)
- Summon (refactor)
- Admin (refactor)
- Axis360 (new dev)
- Aspen API (part of Admin) (new dev)
- SideLoads (new dev)

It also includes a refactor to use one unique usage-graph.tpl file.

Before testing:

1- ensure to have some data in:
     - aspen_usage, user_ils_usage, user_summon_usage, and ils_record_usage 
     (to test that the refactor to one usage graph template file did not break functionality 
     in Admin, Summon or ILS)
     - user_axis360_usage, axis360_record_usage, and axis360_stats
     (to test Axis360 (aka Boundless) graphs and CSV)
     - user_sideload_usage table, sideload_record_usage table
     (ensure to mock at least 2 sideloads instances as a major part of the graph functionality is the ability
     to display the data relevant to one instance)
     - api_usage
     (to test Aspen API graph and CSV)

 2- in Aspen Administration > System Administration > Modules, enable: 
     - Summon
     - Axis360
     This is to allow their dashboards to be displayed. If mock data has been added to Aspen db, then there 
     should be no need for further set up
 
 
Test plan (in essence, a combination of [https://github.com/PTFS-Europe/aspen-discovery/pull/110](https://github.com/PTFS-Europe/aspen-discovery/pull/110) and [https://github.com/PTFS-Europe/aspen-discovery/pull/96](https://github.com/PTFS-Europe/aspen-discovery/pull/96).
 
- once logged in as an Aspen admin, navigate to Aspen Administration > < section name > > Dashboard
- hover over the various links (to the right of each header), check that the title matches
- navigate to the various usage graphs within
- notice that the data on the graph matches the data on the dashboard
- notice the added 'Export as CSV' button, and the accompanying warning message
- click 'Export as CSV'
- if there is a chance the data has changed, refresh the page
- check that the file has been downloaded
- check that the values within the CSV match the ones in 'raw data table'
